### ISSUE TITLE
New declarations posts return 422 "0107189b:3: Cannot delete ephemeral object" when using fqdn addressdiscovery 

### ISSUE THREAD

**Issue by kkloos1** [F5Networks/f5-appsvcs-extension#212]
### Environment
 * Application Services Version: 3.16. 0
 * BIG-IP Version: BIG-IP 13.1.1.3 Build 0.0.1 Point Release 3

### Summary
When I have a declaration that includes pool members that have addressDiscovery set to FQDN, only the initial POST succeeds. All additional attempts to update the declaration fail with the following exception:
 ```
   "results": [
        {
            "code": 422,
            "message": "declaration failed",
            "response": "0107189b:3: Cannot delete ephemeral object: /TEST/TestApp/test_pool_80: /TEST/example.com.",
            "host": "localhost",
            "tenant": "TEST",
            "runTime": 2582
        }
    ],
```

The object being updated doesn't even have to be within the scope of the pool member definition, I initially came across this when changing the redirect80 attribute to false in my Service_HTTPS object included in the same declaration. 

I believe the error gets triggered because when AS3 is attempting to overwrite the current declaration  it tries to delete the current objects defined before replacing them with the new declaration version, when it gets to the pool members section I think AS3 dynamically generates a list of pool members that need to be removed based on what is currently defined in  the pool at execution time. In my AS3 declaration example I only define two FQDN nodes, however these nodes will generate an additional 2 ephemeral nodes that map to the IPs returned to be BigIP from our DNS servers. Both the FQDN and ephemeral nodes show up in the pool (4 total) even though I'm only defining two in my declaration. I believe these ephemeral nodes are immutable and changes can only be inherited from the FQDN node the ephemeral node is generated from. I believe AS3 is trying to delete both the FQDN and ephemeral nodes as opposed to just the FQDN nodes which is triggering the exception.

To get around this bug I have to use tmsh or the web UI to manually remove the pool members from my pool, then re-post the declaration. No changes are made to my AS3 declaration, just manually removing the members from the pool with the same declaration prevents AS3 from triggering this exception. Prior to manually removing the pool members this is how the pool is defined in the conf file (note that there are 4 pools instead of 2):

```
ltm pool /TEST/TestApp/test_pool_80 {
    members {
        /TEST/example.com:http {
            fqdn {
                autopopulate enabled
                name example.com
            }
            metadata {
                source {
                    value declaration
                }
            }
            state fqdn-up
        }
        /TEST/example2.com:http {
            fqdn {
                autopopulate enabled
                name example2.com
            }
            metadata {
                source {
                    value declaration
                }
            }
            state fqdn-up
        }
        _auto_192.168.1.1:http {
            address 192.168.1.1
            ephemeral true
            session monitor-enabled
            state down
            fqdn {
                autopopulate enabled
                name example2.com
            }
        }
        _auto_192.168.1.2:http {
            address 192.168.1.2
            ephemeral true
            session monitor-enabled
            state down
            fqdn {
                autopopulate enabled
                name example.com
            }
        }
    }
    min-active-members 1
    monitor min 1 of { http }
    partition TEST
}
```

### Steps To Reproduce
1. Submit the following declaration:
```json
{
    "class": "AS3",
    "action": "deploy",
    "persist": true,
    "declaration": {
        "class": "ADC",
        "schemaVersion": "3.0.0",
        "TEST": {
            "class": "Tenant",
            "TestApp": {
                "class": "Application",
                "template": "generic",
                "test_vs_80": {
					"class": "Service_HTTP",
                    "virtualAddresses": [
                        "10.0.0.1"
                    ],
                    "pool": "test_pool_80"
                },
                "test_pool_80": {
                    "class": "Pool",
                    "monitors": [
                        "http"
                    ],
                    "members": [
                        {
                        	"addressDiscovery": "fqdn",
                            "autoPopulate": true,
                            "servicePort": 80,
                            "hostname": "example.com"
                        },
                        {
                        	"addressDiscovery": "fqdn",
                            "autoPopulate": true,
                            "servicePort": 80,
                            "hostname": "example2.com"
                        }
                    ]
                }
            }
        }
    }
}
```

2. The initial request is successful, returns a 200 response and the VS + Pool are defined on the BigIP

3. Submit an additional  AS3 POST for this declaration, the POST doesn't even have to include changes in the declaration.

4. Observe the following error response:
```json
        {
            "code": 422,
            "message": "declaration failed",
            "response": "0107189b:3: Cannot delete ephemeral object: /TEST/TestApp/test_pool_80: /TEST/example2.com.",
            "host": "localhost",
            "tenant": "TEST",
            "runTime": 1824
        }
```

### Expected Behavior
I would expect that AS3 would be able to successfully know the order of operations in order to delete and re-define FQDN nodes on a BigIP system. Note that this bug doesn't just impact the ability to change fqdn definitions within a declaration, it prevents the ability to make any changes to a declaration at all if it includes an fqdn node.

### Actual Behavior
BigIP returns a 422 0107189b:3: Cannot delete ephemeral object response because AS3 is attempting to delete an ephemeral node instead of just the 2 parent FQDN nodes. As a result the new declaration I'm posting doesn't get committed.

**Comment by dstokesf5** [comment]
This looks like a duplicate of #204.

**Comment by kkloos1** [comment]
@dstokesf5 this definitely is a duplicate of that issue, sorry about that. From looking at that issues comments it looks like it's been triaged, is there any indication on when the fix will be released?

**Comment by dstokesf5** [comment]
#204 will be fixed in 3.17.0, which should be released around the end of January.
